components/raftstore-v2/src/operation/command/admin/compact_log.rs

```diff
@@ -313,7 +313,13 @@ impl<EK: KvEngine, ER: RaftEngine> Peer<EK, ER> {
             let applied = self.storage().apply_state().get_applied_index();
             let total_cnt = applied - self.storage().entry_storage().first_index() + 1;
             let remain_cnt = applied - compact_index;
-            self.update_approximate_raft_log_size(|s| s * remain_cnt / total_cnt);
+            self.update_approximate_raft_log_size(|s| {
```

Comment:
This is actually the wrong place to update approximate raft log size. Instead, it should be handled in `on_apply_res_compact_log` only and using the compact index from the apply res.